### PR TITLE
fix:  removed unused variables

### LINKED ISSUE #7442: (JS-0128) Should not have unused variables
## Description
Found variables that are declared but not used anywhere.

## Occurrences
There are 247 occurrences of this issue in the repository.

See all occurrences on DeepSource &rarr; [deepsource.io/gh/appsmithorg/appsmith/issue/JS-0128/occurrences/](https://deepsource.io/gh/appsmithorg/appsmith/issue/JS-0128/occurrences/)

### PR DESCRIPTION
**Pull request by gorlenavinkumar**
### Description

Added missing variable usage in the codebase to address the issue of 247 occurrences of declared but unused variables. This pull request aims to enhance the codebase's cleanliness and efficiency by ensuring that all declared variables are utilized appropriately.

**PR fixes following issue(s)**

Fixes #7442 

**Type of change**

Bug fix (non-breaking change which fixes an issue)

### Testing

How Has This Been Tested?
 

- [ ] Manual

### CODE REVIEW

File: app/client/packages/design-system/headless/src/components/Icon/src/Icon.tsx [hunk 1]
```diff
@@ -11,7 +11,7 @@ export interface IconProps extends DOMProps, AriaLabelingProps {
   "aria-hidden"?: boolean | "false" | "true";
 }
 
-export function Icon(props: IconProps) {
+export function _Icon(props: IconProps, ref: React.Ref<SVGSVGElement>) {
```
Comment (KelvinOm) [inline app/client/packages/design-system/headless/src/components/Icon/src/Icon.tsx:14]: @gorlenavinkumar Please update your PR, there are a lot of unrelated changes

**Review by KelvinOm — CHANGES_REQUESTED** [review]

**Comment by ayushpahwa** [comment]
@gorlenavinkumar this PR seems to be outdated as compared to release.
